src/core/global.c

```diff
@@ -101,19 +101,6 @@
     the type should be changed to uint32_t or int. */
 CT_ASSERT (NN_MAX_SOCKETS <= 0x10000);
 
-/*  This check is performed at the beginning of each socket operation to make
-    sure that the library was initialised, the socket actually exists, and is
-    a valid socket index. */
-#define NN_BASIC_CHECKS \
-    if (nn_slow (s < 0 || s > NN_MAX_SOCKETS)) {\
-        errno = EBADF;\
-        return -1;\
-        }\
-    if (nn_slow (!self.socks || !self.socks [s])) {\
-        errno = EBADF;\
-        return -1;\
-    }
-
```

Comment:
In a nutshell, if this went away, where did it go? Briefly scanning this changeset it seems straightforward (to me), but can you say in your own words the design decision here?